sound/soc/sof/intel/hda.c

```diff
@@ -530,29 +536,26 @@ int hda_dsp_probe(struct snd_sof_dev *sdev)
 	 * register our IRQ
 	 * let's try to enable msi firstly
 	 * if it fails, use legacy interrupt mode
-	 * TODO: support interrupt mode selection with kernel parameter
-	 *       support msi multiple vectors
+	 * TODO: support msi multiple vectors
 	 */
 #if IS_ENABLED(CONFIG_PCI)
-	ret = pci_alloc_irq_vectors(pci, 1, 1, PCI_IRQ_MSI);
+	if (hda_use_msi && !pci_alloc_irq_vectors(pci, 1, 1, PCI_IRQ_MSI)) {
+		dev_info(sdev->dev, "use msi interrupt mode\n");
+		hdev->irq = pci_irq_vector(pci, 0);
+		/* ipc irq number is the same of hda irq */
+		sdev->ipc_irq = hdev->irq;
+		/* initialised to "false" by kzalloc() */
+		sdev->msi_enabled = true;
+	}
 #endif
-	if (ret < 0) {
+	if (!sdev->msi_enabled) {
```

Comment:
you don't really need a new field that isn't used anywhere else, a local variable would do.

Reply:
@plbossart it isn't new at all, I just changed its type. And yes, it is used in multiple functions, so, no, it cannot be a local variable